packages/xlucene-evaluator/src/parser/functions/geo/polygon.ts

```diff
@@ -0,0 +1,63 @@
+
+// @ts-ignore
+import { polygon, lineString } from '@turf/helpers';
+// @ts-ignore
+import lineToPolygon from '@turf/line-to-polygon';
+import { geoMatcher } from './helpers';
+import { parseGeoPoint } from '../../../utils';
+import * as i from '../../interfaces';
+import { AnyQuery } from '../../../translator';
+
+function validate(params: i.Term[]) {
+    const geoPointsParam = params.find((node) => node.field === 'points');
+    if (geoPointsParam == null) throw new Error('geoPolygon query needs to specify a "points" parameter');
+    if (!Array.isArray(geoPointsParam.value)) throw new Error('points parameter must be an array');
+    // @ts-ignore we are ignoreing util we have a better story around list expressions
```

Comment:
Using`ts-ignore` is really bad practice event for these issues, use `any` in only the spots you need it. If you don't, the compiler will skip over any possible issue in the next line, instead just a particular value.

```suggestion
    // @ts-ignore we are ignoreing util we have a better story around list expressions
```

Reply:
removed